server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataManager.java

```diff
@@ -993,7 +1000,7 @@ public List<Interval> inTransaction(Handle handle, TransactionStatus status)
                 .createQuery(
                     StringUtils.format(
                         "SELECT start, %2$send%2$s FROM %1$s WHERE dataSource = :dataSource AND "
-                        + "%2$send%2$s <= :end AND used = false ORDER BY start, %2$send%2$s",
+                        + "%2$send%2$s <= :end AND used = false AND (used_flag_last_updated IS NULL or used_flag_last_updated <= :used_flag_last_updated) ORDER BY start, %2$send%2$s",
```

Comment:
I think we can get somewhat better behavior in the NULL case by using the created_date:

```
used = false AND ((used_flag_last_updated IS NULL AND created_date <= :used_flag_last_updated) OR used_flag_last_updated <= :used_flag_last_updated)
```

A little closer to the behavior we'd get if the `used_flag_last_updated` was actually populated.